step-core/src/main/java/step/expressions/GroovyPool.java

```diff
@@ -42,8 +49,16 @@ public GroovyPool(GroovyPoolFactory groovyPoolFactory, int poolMaxTotal, int poo
 			pool.setMaxTotalPerKey(poolMaxTotalPerKey);
 			pool.setMaxIdlePerKey(poolMaxIdlePerKey);
 			pool.setBlockWhenExhausted(true);
-			pool.setTimeBetweenEvictionRunsMillis(30000);
-			pool.setMinEvictableIdleTimeMillis(-1);
+			pool.setTimeBetweenEvictionRuns(Duration.ofMillis(30000));;
+			pool.setMinEvictableIdle(Duration.ofMillis(-1));
+			if (monitoringIntervalSeconds != null) {
+				scheduler = Executors.newScheduledThreadPool(1, r -> {

```

Comment:
new ThreadFactoryBuilder()
					.setNameFormat("groovy-pool-monitor-%d")
					.setDaemon(false)
					.build() from guava would do this. 

The convention seems to use lower case names: groovy-pool-monitor